crates/router/src/core/payments.rs

```diff
@@ -1554,10 +1554,7 @@ pub fn should_call_connector<Op: Debug, F: Clone>(
             !matches!(
                 payment_data.payment_intent.status,
                 storage_enums::IntentStatus::Failed | storage_enums::IntentStatus::Succeeded
-            ) && payment_data
-                .connector_response
-                .authentication_data
-                .is_none()
+            ) && payment_data.payment_attempt.authentication_data.is_none()
```

Comment:
for this case to work, don't we need to backfill payment_attempt table?

Reply:
Yeah. This came from the previous PR , forgot to discard it. Will remove it now.